### PR TITLE
Extend ArcTable.Join logic 

### PR DESCRIPTION
**Pull request by Freymaurer**
Give option to give index where to join table.

✅  Added tests (3)

### CODE REVIEW

File: src/ISA/ISA/ArcTypes/ArcTable.fs [hunk 1]
```diff
@@ -503,12 +503,17 @@ type ArcTable(name: string, headers: ResizeArray<CompositeHeader>, values: Syste
...
     /// </summary>
+    /// <param name="index">If not set default to append. -1 will also append.</param>
     /// <param name="table">The table to join to this table.</param>
     /// <param name="joinOptions">Can add only headers, header with unitized cell information, headers with values.</param>
     /// <param name="forceReplace">if set to true will replace unique columns.</param>
-    member this.Join(table:ArcTable, ?joinOptions: TableJoinOptions, ?forceReplace: bool, ?SkipFillMissing) : unit =
+    member this.Join(table:ArcTable, ?index: int, ?joinOptions: TableJoinOptions, ?forceReplace: bool, ?skipFillMissing) : unit =
         let joinOptions = defaultArg joinOptions TableJoinOptions.Headers
         let forceReplace = defaultArg forceReplace false
+        let skipFillMissing = defaultArg skipFillMissing true
```
Comment (HLWeil) [inline src/ISA/ISA/ArcTypes/ArcTable.fs:513]: Shouldn't the default value be `false`?

Comment (Freymaurer) [reply src/ISA/ISA/ArcTypes/ArcTable.fs:513]: Don't know, you had it as true:
image

Comment (HLWeil) [reply src/ISA/ISA/ArcTypes/ArcTable.fs:513]: The `not` makes the difference here. You could read my code as follows: `Only if skipFillMissing is set to true, do not fill missing cells`

Comment (HLWeil) [reply src/ISA/ISA/ArcTypes/ArcTable.fs:513]: The default case being `fill missing cells`

Comment (Freymaurer) [reply src/ISA/ISA/ArcTypes/ArcTable.fs:513]: The new one also has `true` with `not`?

Comment (Freymaurer) [reply src/ISA/ISA/ArcTypes/ArcTable.fs:513]: ah wait i think i get it. Because yours defaulted to None, it would not match

**Review by HLWeil — APPROVED** [review]

**Pull request merged**